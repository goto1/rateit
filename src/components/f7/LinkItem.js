import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import Icon from "./Icon";

const Content = styled.div`padding-right: 15px;`;

const Inner = styled.div`
  flex-direction: column;
  padding-right: 0 !important;
  position: relative;
`;

const Arrow = styled(Icon)`
  color: #C7C7CC;
  position: absolute !important;
  right: 0;
  width: 17px;
`;

const LinkItem = ({ url, children }) =>
  <li>
    <a href={url}>
      <Content className="item-content">
        <Inner className="item-inner">
          <Arrow material="keyboard_arrow_right" />
          {children}
        </Inner>
      </Content>
    </a>
  </li>;

LinkItem.propTypes = {
  url: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired
};

export default LinkItem;
