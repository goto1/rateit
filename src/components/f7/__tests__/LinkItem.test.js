import React from "react";
import LinkItem from "../LinkItem";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import "jest-enzyme";

describe("LinkItem", () => {
  const HelloWorld = <div>HelloWorld!</div>;

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <LinkItem url="/home">
          {HelloWorld}
        </LinkItem>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("contains `HelloWorld` component", () => {
    const wrapper = shallow(
      <LinkItem url="/home">
        {HelloWorld}
      </LinkItem>
    );
    expect(wrapper).toContainReact(HelloWorld);
  });
});
