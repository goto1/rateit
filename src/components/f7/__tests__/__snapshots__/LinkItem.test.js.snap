// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkItem renders correctly 1`] = `
<li>
  <a
    href="/home"
  >
    <div
      className="item-content sc-bdVaJa lfVYTe"
    >
      <div
        className="item-inner sc-bwzfXH fvZPbp"
      >
        <i
          className="material-icons icon sc-htpNat bNzvmb"
        >
          keyboard_arrow_right
        </i>
        <div>
          HelloWorld!
        </div>
      </div>
    </div>
  </a>
</li>
`;
